coalib/results/TextPosition.py

```diff
@@ -7,20 +7,35 @@
 class TextPosition:
 
     @enforce_signature
-    def __init__(self, line: (int, None)=None, column: (int, None)=None):
+    def __init__(self,
+                 line: (int, None)=None,
+                 column: (int, None)=None,
+                 text: (tuple, list, None)=None,
+                 position: (int, None)=None):
         """
         Creates a new TextPosition object that represents the position inside
         a string with line/column numbers.
 
-        :param line:        The line in file or None, the first line is 1.
-        :param column:      The column indicating the character. The first one
-                            in a line is 1.
-        :raises TypeError:  Raised when line or columns are no integers.
-        :raises ValueError: Raised when a column is set but line is None.
+        :param line:            The line in file or None, the first line is 1.
+        :param column:          The column indicating the character. The first
+                                one in a line is 1.
+        :param text:            The text containing the character.
+        :param position:        Position identifying the index of character
+                                in text.
+        :raises TypeError:      Raised when line or columns are no integers.
+        :raises ValueError:     Raised when a column is set but line is None.
+        :raises AssertionError: Raised when line/column do not match
+                                corresponding text/position.
         """
         if line is None and column is not None:
             raise ValueError("A column can only be set if a line is set.")
 
+        if text and position and line and column:
+            assert (line, column) == TextPosition.from_position(
+                                            text, position, create_object=False)
```

Comment:
there's no probem right? we've had people storing line and column with no reference. in that case we just trust the user :P ? like always?  we have no way of finding out if its correct anyways